packages/application-config/src/process-config.ts

```diff
@@ -37,9 +42,21 @@ const processConfig = ({
     processEnv.MC_APP_ENV ?? processEnv.NODE_ENV ?? 'development';
   const isProd = getIsProd(processEnv);
 
-  // TODO: handle legacy configs for backwards compatibility
-
+  // Read first the new config file. If none is found, attempt to load
+  // the config from the `env.json` and `headers.json` files.
   const loadedAppConfig = loadConfig();
+  if (!loadedAppConfig && deprecatedOptions) {
+    console.warn(
+      `No custom application config found, attempting to load the config from env.json and headers.json.`
+    );
+    const loadedDeprecatedAppConfig = loadDeprecatedConfig(deprecatedOptions);
+    if (!loadedDeprecatedAppConfig) {
+      throw new Error(`No configuration for the Custom Application found.`);
+    }
+    // Return the legacy config as-is, no need to transform it.
+    return loadedDeprecatedAppConfig;
+  }
```

Comment:
This is the logic where we try to load the deprecated files, in case the new config file was not found.

If you have other suggestions for implementing this logic, please let me know.